build-tools/packages/build-cli/src/commands/modify/lockfile.ts

```diff
@@ -0,0 +1,96 @@
+ */
+
+import { updatePackageJsonFile } from "@fluidframework/build-tools";
+import { Flags } from "@oclif/core";
+import execa from "execa";
+
+import { BaseCommand } from "../../base";
+import { releaseGroupFlag } from "../../flags";
+
+/**
+ * Updates the version of a dependency in the lockfile.
+ *
+ * @remarks
+ * This command is primarily used manually to update transitive dependencies when we need to address CVEs.
+ *
+ * Note that this applies to all packages in the specified release group.
+ */
+export default class UpdateDependencyInLockfileCommand extends BaseCommand<
+	typeof UpdateDependencyInLockfileCommand
+> {
+	static readonly summary =
+		"Updates a dependency in the lockfile to the latest version of a specified semver range.";
+
+	static readonly description =
+		`Note that if the version passed in to the command is not within the range of versions ` +
+		`naturally accepted by the packages that depend on it, after this command runs the lockfile might not reflect the ` +
+		`version that was passed in, but the latest version that complies with the semver range declared by the dependent packages.`;
+
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		releaseGroup: releaseGroupFlag({ required: true }),
+		dependencyName: Flags.string({
+			description: "Name of the dependency (npm package) to update",
+			required: true,
+		}),
+		version: Flags.string({
+			description: "Semver range specifier to use when updating the dependency.",
+			required: true,
+		}),
+	};
+
+	public async run(): Promise<void> {
+		const context = await this.getContext();
+		const releaseGroup = context.repo.releaseGroups.get(this.flags.releaseGroup);
+
+		if (releaseGroup === undefined) {
+			// exits the process
+			this.error(`Can't find release group: ${this.flags.releaseGroup}`, { exit: 1 });
+		}
+
+		// Add override to package.json
+		this.info(
+			`Adding pnpm override for ${this.flags.dependencyName}: ${this.flags.version} to package.json`,
+		);
+		updatePackageJsonFile(releaseGroup.directory, (json) => {
+			if (json.pnpm === undefined) {
+				json.pnpm = {};
+			}
+			if (json.pnpm.overrides === undefined) {
+				json.pnpm.overrides = {};
+			}
+			const currentOverrideVersion = json.pnpm.overrides[this.flags.dependencyName] as
+				| string
+				| undefined;
+			if (currentOverrideVersion !== undefined) {
+				this.error(
+					`A pnpm override for the specified dependency already exists (${this.flags.dependencyName}: "${currentOverrideVersion}". Cannot continue.`,
+					{ exit: 1 },
+				);
+			}
+			json.pnpm.overrides[this.flags.dependencyName] = this.flags.version;
+		});
+
+		// Update lockfile
+		this.info(`Updating lockfile`);
+		await execa(`pnpm`, [`install`, `--no-frozen-lockfile`], {
+			cwd: releaseGroup.directory,
+		});
+
+		// Remove override after install
+		this.info(`Restoring package.json to original state`);
+		await execa(`git`, [`restore`, `--source=HEAD`, `package.json`], {
```

Comment:
This is fine, but we've been trying to reduce exec usage overall, preferring to go through a simple-git wrapper. Not worth doing in this case because consistency with the other calls is more important. Just noting for future.